File: app/src/main/java/z/nova/rifmobolgame/PlayGame.java
package z.nova.rifmobolgame;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.view.Window;
import android.view.WindowManager;
import android.widget.Button;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.GridLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.yodo1.mas.banner.Yodo1MasBannerAdListener;
import com.yodo1.mas.banner.Yodo1MasBannerAdView;
import com.yodo1.mas.error.Yodo1MasError;

import java.util.ArrayList;

import z.nova.rifmobolgame.Model.LevelAdapter;
import z.nova.rifmobolgame.Model.TextLevel;
import z.nova.rifmobolgame.singlplayer.Level1;
import z.nova.rifmobolgame.singlplayer.Level10;
import z.nova.rifmobolgame.singlplayer.Level11;
import z.nova.rifmobolgame.singlplayer.Level12;
import z.nova.rifmobolgame.singlplayer.Level13;
import z.nova.rifmobolgame.singlplayer.Level14;
import z.nova.rifmobolgame.singlplayer.Level15;
import z.nova.rifmobolgame.singlplayer.Level16;
import z.nova.rifmobolgame.singlplayer.Level17;
import z.nova.rifmobolgame.singlplayer.Level18;
import z.nova.rifmobolgame.singlplayer.Level19;
import z.nova.rifmobolgame.singlplayer.Level2;
import z.nova.rifmobolgame.singlplayer.Level20;
import z.nova.rifmobolgame.singlplayer.Level21;
import z.nova.rifmobolgame.singlplayer.Level22;
import z.nova.rifmobolgame.singlplayer.Level23;
import z.nova.rifmobolgame.singlplayer.Level24;
import z.nova.rifmobolgame.singlplayer.Level25;
import z.nova.rifmobolgame.singlplayer.Level26;
import z.nova.rifmobolgame.singlplayer.Level27;
import z.nova.rifmobolgame.singlplayer.Level28;
import z.nova.rifmobolgame.singlplayer.Level29;
import z.nova.rifmobolgame.singlplayer.Level3;
import z.nova.rifmobolgame.singlplayer.Level30;
import z.nova.rifmobolgame.singlplayer.Level31;
import z.nova.rifmobolgame.singlplayer.Level32;
import z.nova.rifmobolgame.singlplayer.Level33;
import z.nova.rifmobolgame.singlplayer.Level34;
import z.nova.rifmobolgame.singlplayer.Level35;
import z.nova.rifmobolgame.singlplayer.Level4;
import z.nova.rifmobolgame.singlplayer.Level5;
import z.nova.rifmobolgame.singlplayer.Level6;
import z.nova.rifmobolgame.singlplayer.Level7;
import z.nova.rifmobolgame.singlplayer.Level8;
import z.nova.rifmobolgame.singlplayer.Level9;

public class PlayGame extends AppCompatActivity {

    Yodo1MasBannerAdView bannerAdView;
    ArrayList<TextLevel> textLevels = new ArrayList<TextLevel>();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.play_game);

        bannerAdView = findViewById(R.id.adView);
        bannerAdView.setAdListener(new Yodo1MasBannerAdListener() {
            @Override
            public void onBannerAdLoaded(Yodo1MasBannerAdView bannerAdView) {
                // Code to be executed when an ad finishes loading.
            }

            @Override
            public void onBannerAdFailedToLoad(Yodo1MasBannerAdView bannerAdView, @NonNull Yodo1MasError error) {
                // Code to be executed when an ad request fails.
            }

            @Override
            public void onBannerAdOpened(Yodo1MasBannerAdView bannerAdView) {
                // Code to be executed when an ad opens an overlay that
                // covers the screen.
            }

            @Override
            public void onBannerAdFailedToOpen(Yodo1MasBannerAdView bannerAdView, @NonNull Yodo1MasError error) {
                // Code to be executed when an ad open fails.
            }

            @Override
            public void onBannerAdClosed(Yodo1MasBannerAdView bannerAdView) {
                // Code to be executed when the user is about to return
                // to the app after tapping on an ad.
            }
        });
        bannerAdView.loadAd();

        //Yodo1

        Window w = getWindow();
        w.setFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN, WindowManager.LayoutParams.FLAG_FULLSCREEN);

        Button button_back = (Button) findViewById(R.id.button_back);
        button_back.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                try {
                    Intent intent = new Intent(PlayGame.this, GameMenu.class);
                    startActivity(intent); finish();
                }catch (Exception e){
                //Пусто
                }
            }
        });

        setInitialData();
        RecyclerView recyclerView = findViewById(R.id.list);
        int numberofColums = 5;

        LevelAdapter.ListItemClickListener levelClickListener  = new LevelAdapter.ListItemClickListener() {
            @Override
            public void onListItemClick(int clickItemIndex) {
                Intent intent;
                switch (clickItemIndex){
                    case 0:
                        intent = new Intent(PlayGame.this, Level1.class);
                        startActivity(intent);
                        break;
                    case 1:
                        intent = new Intent(PlayGame.this, Level2.class);
                        startActivity(intent);
                        break;
                    case 2:
                    intent = new Intent(PlayGame.this, Level3.class);
                    startActivity(intent);
                    break;
                    case 3:
                        intent = new Intent(PlayGame.this, Level4.class);
                        startActivity(intent);
                        break;
                    case 4:
                    intent = new Intent(PlayGame.this, Level5.class);
                    startActivity(intent);
                    break;
                    case 5:
                        intent = new Intent(PlayGame.this, Level6.class);
                        startActivity(intent);
                        break;
                    case 6:
                        intent = new Intent(PlayGame.this, Level7.class);
                        startActivity(intent);
                        break;
                    case 7:
                        intent = new Intent(PlayGame.this, Level8.class);
                        startActivity(intent);
                        break;
                    case 8:
                        intent = new Intent(PlayGame.this, Level9.class);
                        startActivity(intent);
                        break;
                    case 9:
                        intent = new Intent(PlayGame.this, Level10.class);
                        startActivity(intent);
                        break;
                    case 10:
                        intent = new Intent(PlayGame.this, Level11.class);
                        startActivity(intent);
                        break;
                    case 11:
                        intent = new Intent(PlayGame.this, Level12.class);
                        startActivity(intent);
                        break;
                    case 12:
                        intent = new Intent(PlayGame.this, Level13.class);
                        startActivity(intent);
                        break;
                    case 13:
                        intent = new Intent(PlayGame.this, Level14.class);
                        startActivity(intent);
                        break;
                    case 14:
                        intent = new Intent(PlayGame.this, Level15.class);
                        startActivity(intent);
                        break;
                    case 15:
                        intent = new Intent(PlayGame.this, Level16.class);
                        startActivity(intent);
                        break;
                    case 16:
                        intent = new Intent(PlayGame.this, Level17.class);
                        startActivity(intent);
                        break;
                    case 17:
                        intent = new Intent(PlayGame.this, Level18.class);
                        startActivity(intent);
                        break;
                    case 18:
                        intent = new Intent(PlayGame.this, Level19.class);
                        startActivity(intent);
                        break;
                    case 19:
                        intent = new Intent(PlayGame.this, Level20.class);
                        startActivity(intent);
                        break;
                    case 20:
                        intent = new Intent(PlayGame.this, Level21.class);
                        startActivity(intent);
                        break;
                    case 21:
                        intent = new Intent(PlayGame.this, Level22.class);
                        startActivity(intent);
                        break;
                    case 22:
                        intent = new Intent(PlayGame.this, Level23.class);
                        startActivity(intent);
                        break;
                    case 23:
                        intent = new Intent(PlayGame.this, Level24.class);
                        startActivity(intent);
                        break;
                    case 24:
                        intent = new Intent(PlayGame.this, Level25.class);
                        startActivity(intent);
                        break;
                    case 25:
                        intent = new Intent(PlayGame.this, Level26.class);
                        startActivity(intent);
                        break;
                    case 26:
                        intent = new Intent(PlayGame.this, Level27.class);
                        startActivity(intent);
                        break;
                    case 27:
                        intent = new Intent(PlayGame.this, Level28.class);
                        startActivity(intent);
                        break;
                    case 28:
                        intent = new Intent(PlayGame.this, Level29.class);
                        startActivity(intent);
                        break;
                    case 29:
                        intent = new Intent(PlayGame.this, Level30.class);
                        startActivity(intent);
                        break;
                    case 30:
                        intent = new Intent(PlayGame.this, Level31.class);
                        startActivity(intent);
                        break;
                    case 31:
                        intent = new Intent(PlayGame.this, Level32.class);
                        startActivity(intent);
                        break;
                    case 32:
                        intent = new Intent(PlayGame.this, Level33.class);
                        startActivity(intent);
                        break;
                    case 33:
                        intent = new Intent(PlayGame.this, Level34.class);
                        startActivity(intent);
                        break;
                    case 34:
                        intent = new Intent(PlayGame.this, Level35.class);
                        startActivity(intent);
                        break;

                }
            }
        };

        recyclerView.setLayoutManager(new GridLayoutManager(this, numberofColums));
        LevelAdapter adapter = new LevelAdapter(this, textLevels, levelClickListener);
        recyclerView.setAdapter(adapter);





    }

    private void setInitialData() {
        textLevels.add(new TextLevel("1"));
        textLevels.add(new TextLevel("2"));
        textLevels.add(new TextLevel("3"));
        textLevels.add(new TextLevel("4"));
        textLevels.add(new TextLevel("5"));
        textLevels.add(new TextLevel("6"));
        textLevels.add(new TextLevel("7"));
        textLevels.add(new TextLevel("8"));
        textLevels.add(new TextLevel("9"));
        textLevels.add(new TextLevel("10"));
        textLevels.add(new TextLevel("11"));
        textLevels.add(new TextLevel("12"));
        textLevels.add(new TextLevel("13"));
        textLevels.add(new TextLevel("14"));
        textLevels.add(new TextLevel("15"));
        textLevels.add(new TextLevel("16"));
        textLevels.add(new TextLevel("17"));
        textLevels.add(new TextLevel("18"));
        textLevels.add(new TextLevel("19"));
        textLevels.add(new TextLevel("20"));
        textLevels.add(new TextLevel("21"));
        textLevels.add(new TextLevel("22"));
        textLevels.add(new TextLevel("23"));
        textLevels.add(new TextLevel("24"));
        textLevels.add(new TextLevel("25"));
        textLevels.add(new TextLevel("26"));
        textLevels.add(new TextLevel("27"));
        textLevels.add(new TextLevel("28"));
        textLevels.add(new TextLevel("29"));
        textLevels.add(new TextLevel("30"));
        textLevels.add(new TextLevel("31"));
        textLevels.add(new TextLevel("32"));
        textLevels.add(new TextLevel("33"));
        textLevels.add(new TextLevel("34"));
        textLevels.add(new TextLevel("35"));
    }

    //Системная кнопка "Назад" - начало
    @Override
    public void onBackPressed(){
        try {
            Intent intent = new Intent(PlayGame.this, GameMenu.class);
            startActivity(intent); finish();
        }catch (Exception e){
        //Пусто
        }
    }
    //Системная кнопка "Назад" - конец
}

File: app/src/main/java/z/nova/rifmobolgame/PlayGame.kt
package z.nova.rifmobolgame

import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.Button
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.GridLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.yodo1.mas.banner.Yodo1MasBannerAdListener
import com.yodo1.mas.banner.Yodo1MasBannerAdView
import com.yodo1.mas.error.Yodo1MasError
import z.nova.rifmobolgame.model.LevelAdapter
import z.nova.rifmobolgame.model.LevelAdapter.ListItemClickListener
import z.nova.rifmobolgame.model.TextLevel
import z.nova.rifmobolgame.singlplayer.*

class PlayGame : AppCompatActivity() {
    lateinit var bannerAdView: Yodo1MasBannerAdView
    var textLevels = ArrayList<TextLevel>()
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.play_game)
        bannerAdView = findViewById(R.id.adView)
        bannerAdView.setAdListener(object : Yodo1MasBannerAdListener {
            override fun onBannerAdLoaded(bannerAdView: Yodo1MasBannerAdView) {
                // Code to be executed when an ad finishes loading.
            }

            override fun onBannerAdFailedToLoad(
                bannerAdView: Yodo1MasBannerAdView,
                error: Yodo1MasError
            ) {
                // Code to be executed when an ad request fails.
            }

            override fun onBannerAdOpened(bannerAdView: Yodo1MasBannerAdView) {
                // Code to be executed when an ad opens an overlay that
                // covers the screen.
            }

            override fun onBannerAdFailedToOpen(
                bannerAdView: Yodo1MasBannerAdView,
                error: Yodo1MasError
            ) {
                // Code to be executed when an ad open fails.
            }

            override fun onBannerAdClosed(bannerAdView: Yodo1MasBannerAdView) {
                // Code to be executed when the user is about to return
                // to the app after tapping on an ad.
            }
        })
        bannerAdView.loadAd()

        //Yodo1

        val button_back = findViewById<View>(R.id.button_back) as Button
        button_back.setOnClickListener {
            try {
                val intent = Intent(this@PlayGame, GameMenu::class.java)
                startActivity(intent)
                finish()
            } catch (e: Exception) {
                //Пусто
            }
        }
        setInitialData()
        val recyclerView = findViewById<RecyclerView>(R.id.list)
        val numberofColums = 5
        val levelClickListener = object : ListItemClickListener {
            override fun onListItemClick(clickItemIndex: Int) {
                val intent: Intent
                when (clickItemIndex) {
                    0 -> {
                        intent = Intent(this@PlayGame, Level1::class.java)
                        startActivity(intent)
                    }
                    1 -> {
                        intent = Intent(this@PlayGame, Level2::class.java)
                        startActivity(intent)
                    }
                    2 -> {
                        intent = Intent(this@PlayGame, Level3::class.java)
                        startActivity(intent)
                    }
                    3 -> {
                        intent = Intent(this@PlayGame, Level4::class.java)
                        startActivity(intent)
                    }
                    4 -> {
                        intent = Intent(this@PlayGame, Level5::class.java)
                        startActivity(intent)
                    }
                    5 -> {
                        intent = Intent(this@PlayGame, Level6::class.java)
                        startActivity(intent)
                    }
                    6 -> {
                        intent = Intent(this@PlayGame, Level7::class.java)
                        startActivity(intent)
                    }
                    7 -> {
                        intent = Intent(this@PlayGame, Level8::class.java)
                        startActivity(intent)
                    }
                    8 -> {
                        intent = Intent(this@PlayGame, Level9::class.java)
                        startActivity(intent)
                    }
                    9 -> {
                        intent = Intent(this@PlayGame, Level10::class.java)
                        startActivity(intent)
                    }
                    10 -> {
                        intent = Intent(this@PlayGame, Level11::class.java)
                        startActivity(intent)
                    }
                    11 -> {
                        intent = Intent(this@PlayGame, Level12::class.java)
                        startActivity(intent)
                    }
                    12 -> {
                        intent = Intent(this@PlayGame, Level13::class.java)
                        startActivity(intent)
                    }
                    13 -> {
                        intent = Intent(this@PlayGame, Level14::class.java)
                        startActivity(intent)
                    }
                    14 -> {
                        intent = Intent(this@PlayGame, Level15::class.java)
                        startActivity(intent)
                    }
                    15 -> {
                        intent = Intent(this@PlayGame, Level16::class.java)
                        startActivity(intent)
                    }
                    16 -> {
                        intent = Intent(this@PlayGame, Level17::class.java)
                        startActivity(intent)
                    }
                    17 -> {
                        intent = Intent(this@PlayGame, Level18::class.java)
                        startActivity(intent)
                    }
                    18 -> {
                        intent = Intent(this@PlayGame, Level19::class.java)
                        startActivity(intent)
                    }
                    19 -> {
                        intent = Intent(this@PlayGame, Level20::class.java)
                        startActivity(intent)
                    }
                    20 -> {
                        intent = Intent(this@PlayGame, Level21::class.java)
                        startActivity(intent)
                    }
                    21 -> {
                        intent = Intent(this@PlayGame, Level22::class.java)
                        startActivity(intent)
                    }
                    22 -> {
                        intent = Intent(this@PlayGame, Level23::class.java)
                        startActivity(intent)
                    }
                    23 -> {
                        intent = Intent(this@PlayGame, Level24::class.java)
                        startActivity(intent)
                    }
                    24 -> {
                        intent = Intent(this@PlayGame, Level25::class.java)
                        startActivity(intent)
                    }
                    25 -> {
                        intent = Intent(this@PlayGame, Level26::class.java)
                        startActivity(intent)
                    }
                    26 -> {
                        intent = Intent(this@PlayGame, Level27::class.java)
                        startActivity(intent)
                    }
                    27 -> {
                        intent = Intent(this@PlayGame, Level28::class.java)
                        startActivity(intent)
                    }
                    28 -> {
                        intent = Intent(this@PlayGame, Level29::class.java)
                        startActivity(intent)
                    }
                    29 -> {
                        intent = Intent(this@PlayGame, Level30::class.java)
                        startActivity(intent)
                    }
                    30 -> {
                        intent = Intent(this@PlayGame, Level31::class.java)
                        startActivity(intent)
                    }
                    31 -> {
                        intent = Intent(this@PlayGame, Level32::class.java)
                        startActivity(intent)
                    }
                    32 -> {
                        intent = Intent(this@PlayGame, Level33::class.java)
                        startActivity(intent)
                    }
                    33 -> {
                        intent = Intent(this@PlayGame, Level34::class.java)
                        startActivity(intent)
                    }
                    34 -> {
                        intent = Intent(this@PlayGame, Level35::class.java)
                        startActivity(intent)
                    }
                }
            }
        }
        recyclerView.layoutManager = GridLayoutManager(this, numberofColums)
        val adapter = LevelAdapter(this, textLevels, levelClickListener)
        recyclerView.adapter = adapter
    }

    private fun setInitialData() {
        textLevels.add(TextLevel("1"))
        textLevels.add(TextLevel("2"))
        textLevels.add(TextLevel("3"))
        textLevels.add(TextLevel("4"))
        textLevels.add(TextLevel("5"))
        textLevels.add(TextLevel("6"))
        textLevels.add(TextLevel("7"))
        textLevels.add(TextLevel("8"))
        textLevels.add(TextLevel("9"))
        textLevels.add(TextLevel("10"))
        textLevels.add(TextLevel("11"))
        textLevels.add(TextLevel("12"))
        textLevels.add(TextLevel("13"))
        textLevels.add(TextLevel("14"))
        textLevels.add(TextLevel("15"))
        textLevels.add(TextLevel("16"))
        textLevels.add(TextLevel("17"))
        textLevels.add(TextLevel("18"))
        textLevels.add(TextLevel("19"))
        textLevels.add(TextLevel("20"))
        textLevels.add(TextLevel("21"))
        textLevels.add(TextLevel("22"))
        textLevels.add(TextLevel("23"))
        textLevels.add(TextLevel("24"))
        textLevels.add(TextLevel("25"))
        textLevels.add(TextLevel("26"))
        textLevels.add(TextLevel("27"))
        textLevels.add(TextLevel("28"))
        textLevels.add(TextLevel("29"))
        textLevels.add(TextLevel("30"))
        textLevels.add(TextLevel("31"))
        textLevels.add(TextLevel("32"))
        textLevels.add(TextLevel("33"))
        textLevels.add(TextLevel("34"))
        textLevels.add(TextLevel("35"))
    }

    //Системная кнопка "Назад" - начало
    override fun onBackPressed() {
        try {
            val intent = Intent(this@PlayGame, GameMenu::class.java)
            startActivity(intent)
            finish()
        } catch (e: Exception) {
            //Пусто
        }
    } //Системная кнопка "Назад" - конец
}